### ISSUE TITLE
Create best practice rule to flag elements with role conflict resolution

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#2284]
We should create a best practice rule that flags elements whose role is `none` or `presentation` and which cause the [role conflict resolution to trigger](https://github.com/dequelabs/axe-core/issues/2212). Role conflict resolution is not a well known concept and getting into that situation causes inconsistent results between browsers. It's also not something you should rely on for accessibility. 

The issue probably shouldn't fail but instead return needs review only.

```html
<!-- this should be flagged as incomplete as the user probably 
     expects this to not be a heading when it in fact is -->
<h1 role="none" aria-label="My Heading"></h1>
```

**Comment by WilcoFiers** [comment]
So, to summarise how to approach this:

1. Needs a new rule, which has a selector of [role=none], [role=presentation]
2. Create two new checks, which go in "all"
    - First check just looks if the element is not focusable. It can use dom.isFocusable()
    - Second check should look if the virtualNode has a global ARIA attribute. There is no commons for this, but it's fairly straight forward:

```js
import lookupTable from '../../commons/aria/lookup-table';

lookupTable.globalAttributes.some(attr =>
	virtualNode.hasAttr(attr)
);
```

Make sure to add tests to `test/checks` and to `test/integration/rules`.

**Comment by straker** [comment]
Note: It might be easier to just check if the explicit role matches the semantic role as the semantic role does all the checking for us :)

**Comment by padmavemulapati** [comment]
Verified new best-practice rule: presentation-role-conflict with axe-core4.0.2 latest code:
1. When the rule passes:
it passes with `<div id="pass1" role="presentation">Test</div>` and `<div id="pass2" role="presentation">Test Button</div>`
![image](https://user-images.githubusercontent.com/47409579/95037897-3cf90a00-06ea-11eb-9a50-70eba02b494f.png)
and the details about the pass condition: 
![image](https://user-images.githubusercontent.com/47409579/95037980-86495980-06ea-11eb-87f9-381ca50d12d7.png)

2. When the rule fails:
it fails with  `<li id="violation1" role="presentation" aria-label="My Heading">Hello</li>` and 
`<h1 id="violation2" role="none" aria-label="My Heading">Hello</h1>`

![image](https://user-images.githubusercontent.com/47409579/95037999-9f520a80-06ea-11eb-91b1-1407efa25838.png)

details on failure:
![image](https://user-images.githubusercontent.com/47409579/95038086-ea6c1d80-06ea-11eb-94dc-f93720e51e0b.png)

![image](https://user-images.githubusercontent.com/47409579/95038172-2606e780-06eb-11eb-88dc-24d889f8f83e.png)
